sdk/python/feast/infra/online_stores/contrib/hazelcast_online_store/hazelcast_online_store.py

```diff
@@ -0,0 +1,317 @@
+
+    _client: Optional[HazelcastClient] = None
+    _lock = threading.Lock()
+
+    def _get_client(self, config: HazelcastOnlineStoreConfig):
+        """
+        Establish the client connection to Hazelcast cluster, if not yet created,
+        and return it.
+
+        The established client connection could be Hazelcast Viridian and SSL enabled based on user config.
+
+        Args:
+            config: The HazelcastOnlineStoreConfig for the online store.
+        """
+        if self._client is None:
+            with self._lock:
+                if self._client is None:
+                    if config.discovery_token != "":
+                        self._client = HazelcastClient(
+                            cluster_name=config.cluster_name,
+                            statistics_enabled=True,
+                            ssl_enabled=True,
+                            cloud_discovery_token=config.discovery_token,
+                            ssl_cafile=config.ssl_cafile_path,
+                            ssl_certfile=config.ssl_certfile_path,
+                            ssl_keyfile=config.ssl_keyfile_path,
+                            ssl_password=config.ssl_password,
+                        )
+                    elif config.ssl_cafile_path != "":
+                        self._client = HazelcastClient(
+                            cluster_name=config.cluster_name,
+                            statistics_enabled=True,
+                            ssl_enabled=True,
+                            ssl_cafile=config.ssl_cafile_path,
+                            ssl_certfile=config.ssl_certfile_path,
+                            ssl_keyfile=config.ssl_keyfile_path,
+                            ssl_password=config.ssl_password,
+                        )
+                    else:
+                        self._client = HazelcastClient(
+                            statistics_enabled=True,
+                            cluster_members=config.cluster_members,
+                            cluster_name=config.cluster_name,
+                        )
+        return self._client
+
+    @log_exceptions_and_usage(online_store="hazelcast")
+    def online_write_batch(
+        self,
+        config: RepoConfig,
+        table: FeatureView,
+        data: List[
+            Tuple[EntityKeyProto, Dict[str, ValueProto], datetime, Optional[datetime]]
+        ],
+        progress: Optional[Callable[[int], Any]],
+    ) -> None:
+        online_store_config = config.online_store
+        if not isinstance(online_store_config, HazelcastOnlineStoreConfig):
+            raise HazelcastInvalidConfig(
+                EXCEPTION_HAZELCAST_UNEXPECTED_CONFIGURATION_CLASS
+            )
+
+        client = self._get_client(online_store_config)
+        fv_map = client.get_map(_map_name(config.project, table))
+
+        for entity_key, values, event_ts, created_ts in data:
+            entity_key_str = base64.b64encode(
+                serialize_entity_key(
```

Comment:
For new online stores, we should just be using `entity_key_serialization_version=2`.

Reply:
Replaced the config parameter with version 2 at https://github.com/feast-dev/feast/pull/3523/commits/dad8fc414d91205c63c1048d956cafe9ae316be6.